src/lib/operators/aggregate_hash.cpp

```diff
@@ -550,12 +622,21 @@ void AggregateHash::_aggregate() {
       auto& result_ids = *context->result_ids;
       auto& results = context->results;
 
-      for (ChunkOffset chunk_offset{0}; chunk_offset < input_chunk_size; chunk_offset++) {
-        // Make sure the value or combination of values is added to the list of distinct value(s). Do not cache result
-        // ids as there is no aggregate function that could reuse the cached indexes.
-        get_or_add_result(std::false_type{}, result_ids, results,
-                          get_aggregate_key<AggregateKey>(keys_per_chunk, chunk_id, chunk_offset),
-                          RowID{chunk_id, chunk_offset});
+      if (_use_immediate_key_shortcut) {
+        for (ChunkOffset chunk_offset{0}; chunk_offset < input_chunk_size; chunk_offset++) {
+          // Make sure the value or combination of values is added to the list of distinct value(s). Do not cache result
```

Comment:
I don't see where this comments describes anything here?

Reply:
My mistake. The comment did not match the code. Thanks. New:

> Make sure the value or combination of values is added to the list of distinct value(s). As we are able to use immediate keys, pass true_type so that the combined caching/immediate key code path is enabled in get_or_add_result.

Better?

Reply:
And where do you want to add this comment?

Reply:
Before `get_or_add_result(std::true_type{}`

Reply:
Since only the true/false_type differs in the function calls, I don't get what this line about the distinct values is about.

Reply:
Ah, I see what you mean. Good point. Yes, this comment is better placed in front of `if (_use_immediate_key_shortcut)`.

> Add value or combination of values is added to the list of distinct value(s). This is done by calling get_or_add_result, which adds the corresponding entry in the list of GROUP BY values.

Reply:
👍